test_submission_schema/v1.json

```diff
@@ -0,0 +1,167 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://github.com/canonical/test-submission-schema/blob/main/test_submission_schema/v1.json",
+    "title": "Test Submission Schema",
+    "description": "A submission schema for submitting test plans.",
+    "type": "object",
+    "additionalProperties": false,
+    "required": [
+        "version",
+        "origin",
+        "session_data",
+        "results"
+    ],
+    "properties": {
+        "version": {
+            "const": 1,
+            "description": "The version of the test submission schema."
+        },
+        "origin": {
+            "type": "object",
+            "required": [
+                "name",
+                "version",
+                "packaging"
+            ],
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "description": "The application used to run the test plan."
+                },
+                "version": {
+                    "type": "string",
+                    "description": "The version of the application used to run the test plan."
+                },
+                "packaging": {
+                    "type": "object",
+                    "properties": {
+                        "type": {
+                            "type": "string",
+                            "description": "The type of application. e.g. Snap, Deb."
+                        },
+                        "name": {
+                            "type": "string",
+                            "description": "The name of the application."
+                        },
+                        "version": {
+                            "type": "string",
+                            "description": "The version of the application."
+                        },
+                        "revision": {
+                            "type": "string",
+                            "description": "The revision of the application."
+                        },
+                        "date": {
+                            "type": "string",
+                            "format": "date",
+                            "description": "The package creation date."
+                        }
+                    },
+                    "if": {"properties": {"type": {"const": "snap"}}},
+                    "then": {"required": ["revision"]}
```

Comment:
By default all properties are optional, so I think your IF/ELSE should be a bit more complex and specify the entire list of properties you want to make required.
if snap -> all of them, else -> all \ revision

Reply:
right, fixed